interview-questions.md

```diff
@@ -9,59 +9,79 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: The 'this' in JavaScript is relates to what 'this' is trying to access, such as this.state."enter prop name here". We use 'this' to access that property and use it throughout our code
 
-  Researched answer:
+  Researched answer: The 'this' keyword references a Javascript element depending on the scope or context of its use. 
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a library extension of Javascript. The reason we use React is because it can update the DOM at a much quicker and efficient pace then vanilla Javascript.  
 
-  Researched answer:
+  Researched answer: React is an open-source Javascript library used to build user interfaces specifically for single-page applications. React also gives us the option of creating reusable UI components. React is great for building dynamic and engaging web interfaces. 
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: The required lifecycle method in a React class component is render(). 
 
-  Researched answer:
+  Researched answer: The render() method is the most used lifecycle method. This method is the only required method within a class component in React. This method handles the rendering of your component to the UI. This happens during the 'mounting' and 'updating' of your component. 
+  Below is an example of a basic rander() in React. 
+
+  class Hello extends Component {
+    render(){
+      return <div>Hello {this.props.name}</divs>
+    }
+  }
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: With JSX, we can implement html into our Javascript React code. JSX is a combination of HTML and JavaScript. 
 
-  Researched answer:
+  Researched answer: JSX is a syntax extension to Javascript. This extension gives us the capability of combining Javascript and HTML code. HTML and Javascript are usually written separately but with React, we can create components that contain the two. 
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is used to install files from the internet.
 
-  Researched answer:
+  Researched answer: Yarn is a package manager for your code which allows you to use and share vode with developers all around the world. Yarn is quick, secure and reliable. Yarn can be beneficial in using other developers' solutions to different problems, making it easier to develop software. These solutions made up of code is shared through something called a package or sometimes referred to as a module. 
 
 
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: 
```

Comment:
Try to treat your answer like you are in an interview.  If they asked you this question, you would respond in some way, even if that is letting them know that you do not know, but will research that.  This is a good time though to also tie in what you do know and see if you can make a connection to the vocabulary.  For instance, you know what a function is and you can define what anonymous is separately.